Solution1/BusinessLayer/Command/BestWeatherCommand.cs

```diff
@@ -1,28 +1,29 @@
 ﻿using BusinessLayer.Command.Abstract;
 using BusinessLayer.DTOs;
+using BusinessLayer.DTOs.Enums;
 using BusinessLayer.Services.Abstract;
-using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace BusinessLayer.Command
 {
-    public class BestWeatherCommand : ICommand<Dictionary<bool, IEnumerable<WeatherResponseDTO>>>
+    public class BestWeatherCommand : ICommand<Dictionary<ResponseStatus, IEnumerable<WeatherResponseDTO>>>
     {
         private readonly IWeatherServiсe _weatherServiсe;
         private readonly IEnumerable<string> _cityNames;
+        private readonly CancellationToken _cancellationToken;
 
-        public BestWeatherCommand(IWeatherServiсe weatherServiсe, IEnumerable<string> cityNames)
+        public BestWeatherCommand(IWeatherServiсe weatherServiсe, IEnumerable<string> cityNames, CancellationToken cancellationToken = new CancellationToken())
```

Comment:
noooo

Reply:
Cancelation token generated ONCE on the top level of application, and passed throw all the path

Reply:
+